source/macos/AdaptiveCards/AdaptiveCards/Base/BaseProtocol.swift

```diff
@@ -28,3 +28,31 @@ protocol InputHandlingViewProtocol: NSView {
     var key: String { get }
     var isValid: Bool { get }
 }
+
+protocol SelectActionHandlingProtocol {
+    func setupSelectAction(selectAction: ACSBaseActionElement?, rootView: NSView) -> TargetHandler?
+}
+
+extension SelectActionHandlingProtocol {
+    func setupSelectAction(selectAction: ACSBaseActionElement?, rootView: NSView) -> TargetHandler? {
+        guard let selectAction = selectAction, let rootView = rootView as? ACRView else { return nil }
+        var target: TargetHandler?
+        switch selectAction.getType() {
+        case .openUrl:
+            guard let openURLAction = selectAction as? ACSOpenUrlAction else { break }
+            target = ActionOpenURLTarget(element: openURLAction, delegate: rootView)
+            
+        case .submit:
+            guard let submitAction = selectAction as? ACSSubmitAction else { break }
+            target = ActionSubmitTarget(element: submitAction, delegate: rootView)
+            
+        default:
+            break
+        }
+        
+        if let actionTarget = target {
+            rootView.addTarget(actionTarget)
+        }
+        return target
+    }
+}
```

Comment:
This can be added as a new file under `Base` folder just like `ResourceResolverProtocol`